scan_explorer_service/views/image_proxy.py

```diff
@@ -38,106 +43,137 @@ def image_proxy(path):
-                    remove = urlparse.urlparse(url_for_proxy('proxy.image_proxy', path='')).path
-                    path = path.replace(remove, '')
-                    im_data = image_proxy(path).get_data()
-                    memory_sum += sys.getsizeof(im_data)
-                    current_app.logger.debug(f'Page {n_pages}. Image data: {im_data}')
-                    yield im_data
-                fetch_images_end_time = time.time()
-                fetch_images_time_elapsed = fetch_images_end_time - fetch_images_start_time
-                current_app.logger.debug(f'Fetching images for pages took {fetch_images_time_elapsed:.2f} seconds to complete.')
-
-        current_app.logger.debug('######## Starting image loop ########')
-        loop_start_time = time.time() 
-        response = Response(img2pdf.convert([im for im in loop_images(id, page_start, page_end)]), mimetype='application/pdf')
-        loop_end_time = time.time() 
-        loop_total_time = loop_end_time - loop_start_time
-        current_app.logger.debug(f'Image loop took {loop_total_time:.2f} seconds to complete.')
-        return response
+            query = get_pages(item, session, page_start, page_end, page_limit)
+            
+            for page in query.all():
+                
+                n_pages += 1
+                
+                current_app.logger.debug(f"Generating image for page: {n_pages}") 
+                current_app.logger.debug(f'Id: {page.id}, Volume_page: {page.volume_running_page_num}, memory: {memory_sum}')
+                if n_pages > page_limit:
+                    break
+                if memory_sum > memory_limit:
+                    current_app.logger.debug(f"Memory limit reached: {memory_sum} > {memory_limit}") 
+                    break
+                size = 'full'
+                if dpi != 600:
+                    size = str(int(page.width*scaling))+ ","
+                image_url = page.image_url + "/full/" + size + f"/0/{page.image_color_quality}.tif"
+                path = urlparse.urlparse(image_url).path
+                remove = urlparse.urlparse(url_for_proxy('proxy.image_proxy', path='')).path
+                path = path.replace(remove, '')
+                current_app.logger.debug(f"Getting image data...: {n_pages}") 
+                im_data = image_proxy(path).get_data()
+                current_app.logger.debug(f"Getting image data...: {im_data}") 
+                yield im_data
+        
+        with current_app.session_scope() as session:
+            profiler = cProfile.Profile()
+            profiler.enable()
+
+            item = get_item(session, id) 
+            current_app.logger.debug(f"Item retrieved successfully: {item.id}")
+           
+            if isinstance(item, Article): 
+                current_app.logger.info(f"Item is an article: {item.id}")
+                object_name = f'{item.id}.pdf'
+                full_path = f'pdfs/{object_name}'
```

Comment:
hitting full path with `'object_name'.lower()` might resolve the 404 issues. You'll just want to make sure that other places that use full path don't care if the case changes.